state/indexer/sink/psql/backport.go

```diff
@@ -0,0 +1,78 @@
+package psql
+
+// This file adds code to the psql package that is needed for integration with
+// v0.34, but which is not part of the original implementation.
+
+import (
+	"context"
+	"errors"
+
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/pubsub/query"
+	"github.com/tendermint/tendermint/state/txindex"
+	"github.com/tendermint/tendermint/types"
+)
+
+const (
+	eventTypeBeginBlock = "begin_block"
+	eventTypeEndBlock   = "end_block"
+)
+
+// TxIndexer returns a bridge from es to the Tendermint v0.34 transaction indexer.
+func (es *EventSink) TxIndexer() BackportTxIndexer {
```

Comment:
So that I'm understanding:[ ADR 65](https://github.com/tendermint/tendermint/blob/master/docs/architecture/adr-065-custom-event-indexing.md#L17) was implemented during this release cycle. That ADR merged the `TxIndexder` into the `EventSink` interface. This type exists to make the _new_ `psql` ES look like the old TxIndexer. This, it looks like, exists as a shim between the implementations and to allow some amount of code reuse across 0.34 and 0.35, is that correct?